lektor/markdown.py

```diff
@@ -11,9 +11,18 @@
 
 _markdown_cache = threading.local()
 
+_old_mistune = int(mistune.__version__.split('.', maxsplit=1)[0]) < 2
 
-class ImprovedRenderer(mistune.Renderer):
-    def link(self, link, title, text):
+if _old_mistune:
+    from mistune import Renderer as BaseRenderer
+else:
+    from mistune import HTMLRenderer as BaseRenderer  # pylint: disable=no-name-in-module
+
+
+class ImprovedRenderer(BaseRenderer):
+    def link(self, link, text=None, title=None):  # pylint: disable=arguments-differ, arguments-renamed
+        if _old_mistune:
+            (title, text) = (text, title)
```

Comment:
Note that [mistune 1.x version of link][1] does an
```py
if text is None:
    text = link
```
around here.  The implication is that sometimes this gets called with just a `link` (URL) and no text, in which case we want to display the URL for the text of the link.  I suspect this might be needed sometimes.

[1]: <https://github.com/lepture/mistune/blob/46a65a978b005651dcc54a340e44aef71f9f7e95/mistune/renderers.py#L137>